File: python/sample2.py
#!/usr/bin/env python3
from flask import Flask, session, request, render_template

app = Flask(__name__)

@app.route("/countup", methods=["get", "post"])
def countup():
    # もしリクエストメソッドがPOSTだった、またはセッション情報がセットされていなかった場合
    if (request.method == "POST" or "count" not in session):
        # セッション情報を0で初期化
        session["count"] = 0
    else:
        session["count"] += 1
    return render_template("sample2.html", count=str(session["count"]))

app.secret_key = "がっこうぐらし！"
app.run(host="0.0.0.0", port=8000)

File: go/sample2.go
package main

import (
	"github.com/gin-gonic/gin"
	"github.com/gin-contrib/sessions"
	"net/http"
)

func countup(c *gin.Context) {
	session := sessions.Default(c)
	var count int
	v := session.Get("count")
	if v == nil {
		count = 0
	} else {
		count = v.(int)
		count++
	}
	session.Set("count", count)
	session.Save()
	c.HTML(http.StatusOK, "sample2.html", gin.H{
		"count": count,
	})
}

func countupReset(c *gin.Context) {
	session := sessions.Default(c)
	count := 0
	session.Set("count", count)
	session.Save()
	c.HTML(http.StatusOK, "sample2.html", gin.H{
		"count": count,
	})
}
